internal/state/store.go

```diff
@@ -130,12 +139,21 @@ func IsEmpty(store dbStore) (bool, error) {
 
 // NewStore creates the dbStore of the state pkg.
 func NewStore(db dbm.DB, options StoreOptions) Store {
```

Comment:
What do you think about having two methods: `NewStore` and `NewStoreWithOptions`?

Reply:
This interface was there from before so I would not necessarily change it in this PR although I think it is nicer to do it differently. I did it more elegantly for the blockstore now btw.

Reply:
I agree that the `WithMetrics` solution is nicer. I also have to agree that we can't fix all the code in one go. But maybe it's worth opening a new issue on Anton's comment and we'll get back to it later.

Reply:
Will do, is that ok with you @melekes ?